src/checkInput.js

```diff
@@ -0,0 +1,146 @@
+import merge from 'merge-deep';
+import getMaxValue from 'ml-array-max';
+
+import { sumOfGaussianLorentzians } from './shapes/sumOfGaussianLorentzians';
+import { sumOfGaussians } from './shapes/sumOfGaussians';
+import { sumOfLorentzians } from './shapes/sumOfLorentzians';
+
+export function checkInput(data, peaks, options) {
+  let {
+    shape = { kind: 'gaussian' },
+    optimization = {
+      kind: 'lm',
+    },
+  } = options;
+
+  if (typeof shape.kind !== 'string') {
+    throw new Error('kind should be a string');
+  }
+
+  let kind = shape.kind.toLowerCase().replace(/[^a-z]/g, '');
+
+  let paramsFunc;
+  let defaultParameters;
+  switch (kind) {
+    case 'gaussian':
+      paramsFunc = sumOfGaussians;
+      defaultParameters = {
+        x: {
+          init: (peak) => peak.x,
+          max: (peak) => peak.x + peak.width * 2,
```

Comment:
Based on your experience we should really allow such a shift ? I would rather expect peak.width/2 but you may be right.

Reply:
When the guess is bad e.g the peak picking is failing due to noise data, it is good to have a little bit of freedom. it is the case for TGA data where the curve is not smooth and/or staggered.